### PR TITLE
Add initial BigQuery API access.

### PR DESCRIPTION
**Pull request by jimmc**
This PR implements some basic API calls to BigQuery. At the moment, these are only used from a sample debugging call that gets executed when the Search button is pressed, and which just logs to the console. The next step is to figure out how the user can supply the values that are passed to these functions, and how to display the results.

### CODE REVIEW

File: sources/web/datalab/polymer/modules/GapiManager.ts [hunk 1]
```diff
@@ -34,7 +40,8 @@ class GapiManager {
     // Loads the gapi client library and the auth2 library together for efficiency.
     // Loading the auth2 library is optional here since `gapi.client.init` function will load
     // it if not already loaded. Loading it upfront can save one network request.
-    GapiManager._loadClientId()
+    console.log('== loading gapi');
+    return GapiManager._loadClientId()
```
Comment (yebrahim) [inline sources/web/datalab/polymer/modules/GapiManager.ts]: Add `Async` to the function `loadGapi` if it's now asynchronous.

File: third_party/externs/ts/gapi/bigquery.d.ts [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+      projectId: string;
+      all?: boolean;    // true to list all datasets, including hidden ones
+      filter?: string;  // syntax: "labels.<name>[:<value>]"
+            // multiple filters can be ANDed by connecting with a space
+      maxResults?: number  // (unsigned int)
+      pageToken?: string  // page token as returned by a previous call
+    }
+
+    interface ListDatasetsResponse {
+      /*
```
Comment (yebrahim) [inline third_party/externs/ts/gapi/bigquery.d.ts]: Why are these commented out?

File: third_party/externs/ts/gapi/gapi.d.ts [hunk 3]
```diff
@@ -9,12 +9,24 @@ declare module gapi.client {
...
   }): Promise<void>;
 
+  export function load(name: string, version: string): Promise<void>;
+
   interface RequestOptions {
     path: string;
     params?: any;
   }
 
   export function request(args: RequestOptions): Promise<any>;
```
Comment (yebrahim) [inline third_party/externs/ts/gapi/gapi.d.ts]: Where is this used?

File: sources/web/datalab/polymer/modules/GapiManager.ts (hunk 1 above)
Comment (jimmc) [inline sources/web/datalab/polymer/modules/GapiManager.ts]: Are you proposing that all functions that return a Promise should have a name that ends in Async? I don't think I agree with that (and we have a ton of "Async"s to add if we're going to do it). It sounds kind of like hungarian notation.
Or am I misunderstanding your request?

Comment (yebrahim) [inline sources/web/datalab/polymer/modules/GapiManager.ts]: Yes this was my proposition, I'm also not a huge fan of tying the type system to the naming, but I think it's useful to quickly know the logic flow as I'm calling the function.
How about we stick to an explicit function return type then?

Comment (jimmc) [inline sources/web/datalab/polymer/modules/GapiManager.ts]: I think putting explicit return types on all public async functions is a fine idea. Done (for GapiManager).

**Pull request merged**